apps/prairielearn/src/lib/course-request.ts

```diff
@@ -0,0 +1,107 @@
+import { loadSqlEquiv, queryRow, queryAsync } from '@prairielearn/postgres';
+import { z } from 'zod';
+import { logger } from '@prairielearn/logger';
+import * as Sentry from '@prairielearn/sentry';
+
+import { DateFromISOString, IdSchema } from '../lib/db-types';
+import { createCourseRepoJob } from '../lib/github';
+import { sendCourseRequestMessage } from '../lib/opsbot';
+
+const sql = loadSqlEquiv(__filename);
+
+const JobsRowSchema = z
+  .object({
+    authn_user_id: IdSchema.nullable(),
+    authn_user_name: z.string().nullable(),
+    finish_date: DateFromISOString.nullable(),
+    id: IdSchema,
+    number: z.number(),
+    start_date: DateFromISOString,
+    status: z.string(),
+  })
+  .optional();
+
+const CourseRequestRowSchema = z.object({
+  approved_by_name: z.string().nullable(),
+  first_name: z.string().nullable(),
+  github_user: z.string().nullable(),
+  id: IdSchema,
+  institution: z.string().nullable(),
+  jobs: z.array(JobsRowSchema),
+  last_name: z.string().nullable(),
+  short_name: z.string().nullable(),
+  status: z.enum(['pending', 'approved', 'denied', 'creating', 'failed']),
+  title: z.string().nullable(),
+  user_uid: z.string().nullable(),
+  user_name: z.string().nullable(),
+  work_email: z.string().nullable(),
+});
+
+async function selectCourseRequests(show_all: boolean) {
+  // return await queryRow(sql.get_requests, { show_all }, z.array(CourseRequestRowSchema));
+  return await queryRow(sql.get_requests, { show_all }, z.array(CourseRequestRowSchema));
+}
+
+export async function selectAllCourseRequests() {
+  return await selectCourseRequests(true);
+}
+
+export async function selectPendingCourseRequests() {
+  return await selectCourseRequests(false);
+}
+
+export async function updateCourseRequest(req, res) {
+  let action = req.body.approve_deny_action;
+  if (action === 'deny') {
+    action = 'denied';
+  } else {
+    throw new Error(`Unknown course request action "${action}"`);
+  }
+
+  await queryAsync(sql.update_course_request, {
+    id: req.body.request_id,
+    user_id: res.locals.authn_user.user_id,
+    action,
+  });
+  res.redirect(req.originalUrl);
+}
+
+export async function createCourseFromRequest(req, res) {
+  console.log(res.locals.authn_user.user_id);
```

Comment:
```suggestion
```

Reply:
Removed in [da0388f](https://github.com/PrairieLearn/PrairieLearn/pull/9238/commits/da0388fa6b3f7cf5f55e4845555c3125a768be4b).